DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h

```diff
@@ -0,0 +1,69 @@
+#ifndef CTPPS_CTPPSDigi_CTPPSPixelDigi_h
+#define CTPPS_CTPPSDigi_CTPPSPixelDigi_h
+/**
+ * Persistent digi for CTPPS Pixels.
+ * Author: F.Ferro ferro@ge.infn.it
+ */
+
+#include "FWCore/Utilities/interface/Exception.h"
+
+class CTPPSPixelDigi {
+public:
+
+
+  CTPPSPixelDigi( int packed_value) : theData(packed_value) {}
+
+  CTPPSPixelDigi( int row, int col, int adc) {
+    init( row, col, adc);
+  }
+
+  CTPPSPixelDigi( int chan, int adc) {
+    std::pair<int,int> rc = channelToPixel(chan);
+    init( rc.first, rc.second, adc);
+  }
+
+  CTPPSPixelDigi() : theData(0)  {}
+
+  /// Access to digi information
+  int row() const     {return (theData >> row_shift) & row_mask;}
+  int column() const  {return (theData >> column_shift) & column_mask;} 
+  unsigned short adc() const  {return (theData >> adc_shift) & adc_mask;}
+  uint32_t packedData() const {return theData;}
+
+  static std::pair<int,int> channelToPixel( int ch) {
+    int row = ( ch >> column_width) & row_mask;
```

Comment:
The data are packed the way you say. The channel is defined the other way round rrrrrrrccccccc. It's a legacy from SiPixels and I kept it.